prov/sockets/src/sock_av.c

```diff
@@ -562,13 +562,16 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 	table_sz = sizeof(struct sock_av_table_hdr) +
 		_av->attr.count * sizeof(struct sock_av_addr);
 	
-	if (attr->name) {
-		_av->name = calloc(1, FI_NAME_MAX);
+	if (attr->name && strlen(attr->name) > FI_NAME_MAX) {
+		SOCK_LOG_ERROR("attr name is larger than FI_NAME_MAX\n");
+		ret = -FI_EINVAL;
+		goto err1;
```

Comment:
Since we were allocating memory of size FI_NAME_MAX, I thought we needed to make sure attr->name is not larger than that. I'll remove it then.